src/mantine-core/src/components/Spoiler/Spoiler.module.css

```diff
@@ -1,6 +1,9 @@
 .root {
   position: relative;
-  margin-bottom: rem(24px);
+
+  &[data-has-spoiler] {
+    margin-bottom: var(--spoiler-margin-bottom, rem(24px));
```

Comment:
Do not use CSS properties inside data-* attributes, it will be impossible to override without !important. Please see how other components handle data atrributes.